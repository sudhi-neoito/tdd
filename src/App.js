import logo from "./logo.svg";
import "./App.css";
import { useState } from "react";

const App = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const [error, setError] = useState("");

  const onClickButton = () => {
    setError("");
    if (username === "" || password === "") {
      setError("Invalid Username and Password");
    } else {
      return "saved";
    }
    return "not";
  };

  return (
    <div className="App">
      <div>
        <h2>Testing here</h2>
        <span id="error">{error}</span>
        <div>
          <input
            type="text"
            name="username"
            id="username"
            placeholder="User Name"
            value={username}
            onChange={(event) => setUsername(event.target.value)}
          />
        </div>
        <div>
          <input
            type="password"
            id="password"
            name="password"
            placeholder="Password"
            value={password}
            onChange={(event) => setPassword(event.target.value)}
          />
        </div>
        <div>
          <button id="login-button" onClick={() => onClickButton()}>
            LOGIN
          </button>
        </div>
      </div>
    </div>
  );
};

export default App;
