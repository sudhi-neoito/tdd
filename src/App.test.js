import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

describe("App", () => {
  let container = null;
  beforeEach(() => {
    container = shallow(<App />);
  });

  afterEach(() => {
    container.unmount();
  });

  it("should render correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a div <div />", () => {
    const wrapper = container.find("div").length;
    expect(wrapper).toBeGreaterThanOrEqual(1);
  });

  it("Should render a input <input/>", () => {
    const wrapper = container.find("input").length;
    expect(wrapper).toEqual(2);
  });

  it("Should render a button <button/>", () => {
    const wrapper = container.find("button").length;
    expect(wrapper).toEqual(1);
  });

  it("Should have  a text LOGIN", () => {
    const wrapper = container.find("button").text();
    expect(wrapper).toContain("LOGIN");
  });

  it("has an input filed the user can type username", () => {
    const wrapper = container;
    wrapper.find("input[name='username']").simulate("change", {
      target: {
        value: "username",
      },
    });
    wrapper.update();

    expect(wrapper.find("input[name='username']").props().value).toEqual("username");
  });
  it("has an input filed the user can type password", () => {
    const wrapper = container;
    wrapper.find("input[name='password']").simulate("change", {
      target: {
        value: "password",
      },
    });
    wrapper.update();
    expect(wrapper.find("input[name='password']").props().value).toEqual("password");
  });
  it("has a button to Login User without values", () => {
    const wrapper = container;
    wrapper.find("input[name='username']").simulate("change", {
      target: {
        value: "",
      },
    });
    wrapper.update();
    expect(wrapper.find("input[name='username']").props().value).toEqual("");
    wrapper.find("input[name='password']").simulate("change", {
      target: {
        value: "",
      },
    });
    wrapper.update();
    expect(wrapper.find("input[name='password']").props().value).toEqual("");
    wrapper.find("button").find("#login-button").simulate("click");

    expect(wrapper.find("span").find("#error").text()).toEqual("Invalid Username and Password");
  });

  it("has a button Login with values", () => {
    const wrapper = container;
    wrapper.find("input[name='username']").simulate("change", {
      target: {
        value: "sameera",
      },
    });
    wrapper.update();
    expect(wrapper.find("input[name='username']").props().value).toEqual("sameera");
    wrapper.find("input[name='password']").simulate("change", {
      target: {
        value: "sam123",
      },
    });
    wrapper.update();
    expect(wrapper.find("input[name='password']").props().value).toEqual("sam123");
    expect(wrapper.find("button").find("#login-button").props().onClick()).toEqual("saved");
  });
  it("Should have a button onClick event", () => {
    const wrapper = container.find("button").find("#login-button").props().onClick();

    expect(wrapper).toEqual("not");
  });
});
