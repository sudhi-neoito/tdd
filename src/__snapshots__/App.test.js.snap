// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <div>
    <h2>
      Testing here
    </h2>
    <span
      id="error"
    />
    <div>
      <input
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="User Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div>
      <button
        id="login-button"
        onClick={[Function]}
      >
        LOGIN
      </button>
    </div>
  </div>
</div>
`;
